### PR TITLE
Hide hard-coded commands from the agent

### PR DESCRIPTION
**Pull request by rbren**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below
no changelog needed

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

When working with a GH repo, the agent still pushes to main very aggressively. It seems this is because the session starts with the commands
```
export GITHUB_TOKEN=***
git clone ...
```

Hiding this from the agent makes it behave much better

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Review by tofarr — APPROVED** [review]

File: frontend/src/services/observations.ts [hunk 1]
```diff
@@ -10,6 +10,7 @@ import { addAssistantMessage } from "#/state/chatSlice";
 export function handleObservationMessage(message: ObservationMessage) {
   switch (message.observation) {
     case ObservationType.RUN:
+      if (message.args.hidden) break;
```
Comment (rbren) [inline frontend/src/services/observations.ts:13]:
```suggestion
      if (message.hidden) break;
```

File: frontend/src/components/project-menu/ProjectMenuCard.tsx [hunk 2]
```diff
@@ -37,8 +37,13 @@ export function ProjectMenuCard({
 
   const handlePushToGitHub = () => {
     const rawEvent = {
-      content:
-        "Please create a new branch and commit the changes. Then push them to the remote repository, and open up a pull request using the GitHub API and the token in the GITHUB_TOKEN environment variable",
+      content: `
+Let's push the code to GitHub.
+If we're currently on the openhands-workspace branch, please create a new branch with a descriptive name.
+Commit any changes and push them to the remote repository.
+Finally, pen up a pull request using the GitHub API and the token in the GITHUB_TOKEN environment variable, then show me the URL of the pull request.
```
Comment (enyst) [inline frontend/src/components/project-menu/ProjectMenuCard.tsx:44]:
```suggestion
Finally, open up a pull request using the GitHub API and the token in the GITHUB_TOKEN environment variable, then show me the URL of the pull request.
```

Comment (rbren) [reply frontend/src/components/project-menu/ProjectMenuCard.tsx]: whoops good catch!

File: frontend/src/entry.client.tsx [hunk 3]
```diff
@@ -14,7 +14,7 @@ import "./i18n";
 import store from "./store";
 
 async function prepareApp() {
-  if (process.env.NODE_ENV === "development") {
+  if (false && process.env.NODE_ENV === "development") {
```
Comment (amanape) [inline frontend/src/entry.client.tsx:17]:
```suggestion
  if (process.env.NODE_ENV === "development") {
```

**Review by amanape — APPROVED** [review]

**Pull request merged**